extensions/azure/cosmos/contract-negotiation-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/negotiation/store/CosmosContractNegotiationStoreIntegrationTest.java

```diff
@@ -322,37 +291,30 @@ void nextForState_leasedBySelf() {
 
         // verify that the subsequent call to nextForState does not return the entity
         result = store.nextForState(state.code(), 10);
-
         assertThat(result).isEmpty();
     }
 
     @Test
-    void nextForState_leaseByAnotherExpired() throws InterruptedException {
+    void nextForState_leasedByAnotherExpired() {
         var state = ContractNegotiationStates.CONFIRMED;
         var n = generateNegotiation(state);
         var doc = new ContractNegotiationDocument(n, partitionKey);
-        doc.acquireLease("another-connector", Duration.ofMillis(10));
-        container.createItem(doc);
-
-        Thread.sleep(20); //give the lease time to expire
-
-        var result = store.nextForState(state.code(), 10);
-        assertThat(result).hasSize(1).allSatisfy(neg -> assertThat(neg).usingRecursiveComparison().isEqualTo(n));
-    }
-
-    @Test
-    void nextForState_lockEntity() {
-        var n = generateNegotiation("test-id-lock", ContractNegotiationStates.CONSUMER_OFFERED);
-        var doc = new ContractNegotiationDocument(n, partitionKey);
+        Duration leaseDuration = Duration.ofSeconds(5);
```

Comment:
Does this mean this test takes at least 5 seconds? If so, I think we should reduce this value.

Reply:
yes, this test will take at least 5 sec. How much can we reduce it? 2 sec?

Reply:
I'll try with 2 seconds then

Reply:
Why not bringing it down to some milliseconds?

Reply:
we can have 2 separate tests for that but IMO it makes sense to have it in one test scenario, to be sure the flow is correct